### PR TITLE
step-44: Change the FE space used for p,J

### LINKED ISSUE #11612: Step-44: Use FE_DGP instead of FE_DGPMonomial?
After a discussion with @peterrum, I was compelled to try substituting the `FE_DGPMonomial` used for step-44's pressure and dilatation fields by `FE_DGP`. After re-reading the documentation for these two classes, it appeared to me that `FE_DGP` should work just fine. In order to satisfy the LBB condition, the requirement of the fields is that they are discontinuous, of one degree lower than the displacement field, and contain the complete polynomials of the FE space (see Hughes "The Finite Element Method" p295/296). For example, in 2d a bi-quadratic displacement must be paired with a piece-wise linear (not bilinear) pressure and dilatation. Both the [`FE_DGP`](https://dealii.org/developer/doxygen/deal.II/classFE__DGP.html) and [`FE_DGPMonomial`](https://dealii.org/developer/doxygen/deal.II/classFE__DGPMonomial.html) classes claim to do this, with the main difference between them being the following:
```
The basis functions for this element [FE_DGPMonomial] are chosen to be the monomials listed above. Note that this is the main difference to the FE_DGP class that uses a set of polynomials of complete degree p that form a Legendre basis on the unit square. Thus, there, the mass matrix is diagonal, if the grid cells are parallelograms. The basis here does not have this property; however, it is simpler to compute. On the other hand, this element has the additional disadvantage that the local cell matrices usually have a worse condition number than the ones originating from the FE_DGP element.
```

So it would seem that we can expect both to work, and should observe a difference in condition number of the linear system. I've tested this with order 4 displacement (and corresponding order 3 pressure and displacement fields) and this does seem to be the case. The `FE_DGP` seem to give superior performance.
[step-44-Q(4)_DGP(3)_DGP(3).txt](https://github.com/dealii/dealii/files/5860133/step-44-Q.4._DGP.3._DGP.3.txt)
[step-44-Q(4)_DGPMonomial(3)_DGPMonomial(3).txt](https://github.com/dealii/dealii/files/5860132/step-44-Q.4._DGPMonomial.3._DGPMonomial.3.txt)

Unfortunately, I cannot recall why we chose the `FE_DGPMonomial` element as opposed to the `FE_DGP`. @mac-a, do you have any recollection as to why we did this? I propose that we investigate changing the element type for these discontinuous fields, on the basis of an increase in performance with no apparent decrease in the accuracy of the element.

[step-44.cc.txt](https://github.com/dealii/dealii/files/5860225/step-44.cc.txt)

### PR DESCRIPTION
**Pull request by jppelteret**
Fixes #11612

### CODE REVIEW

**Review by kronbichler — APPROVED** [review]
I agree with this change, this makes a lot of sense (the condition numbers can be hilarious with FE_DGPMonomial on bigger meshes).

File: doc/news/changes/minor/20220426Jean-PaulPelteret [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Improved: The step-44 tutorial now uses FE_DGP elements to discretise the
```
Comment (kronbichler) [inline doc/news/changes/minor/20220426Jean-PaulPelteret:1]: I am not sure if we strive for American English consistently, but let me suggest it anyway :wink:
```suggestion
Improved: The step-44 tutorial now uses FE_DGP elements to discretize the
```

**Comment by bangerth** [comment]
I agree with the change to step-44 as well. For the tests, my preference would have been to duplicate the existing tests, rather than change them. There is value in ensuring that both approaches continue to work, regardless of whether they are the "better" or "worse" way to approach things.

File: doc/news/changes/minor/20220426Jean-PaulPelteret (hunk 1 above)
Comment (jppelteret) [reply doc/news/changes/minor/20220426Jean-PaulPelteret]: Thanks, I've now fixed this.

**Comment by jppelteret** [comment]
Hi @bangerth,

> For the tests, my preference would have been to duplicate the existing tests, rather than change them.

Fair point. I've duplicated just the test `physics/step-44.cc` to `physics/step-44-dgp_monomial.cc` which now gives coverage to both implementations. Does that work for you?

**Comment by jppelteret** [comment]
Hi @kronbichler,

Thanks for the review!

> the condition numbers can be hilarious with FE_DGPMonomial on bigger meshes

I didn't know that it could get that bad. I swear that when we implemented the tutorial we started off with `FE_DGP` but didn't get any results, and then we switched to `FE_DGPMonomial` and suddenly we got a solution. I don't know why that would be, but clearly that's no longer the case.

**Review by bangerth — APPROVED** [review]

**Pull request merged**